src/systems/filecoin_mining/storage_mining/storage_miner_actor.id

```diff
@@ -1,166 +1,99 @@
-    )
-    _updateExpireSectors(currEpoch block.ChainEpoch) [sector.SectorNumber]
-    _updateClearSectorAssert(sectorNo sector.SectorNumber)
-    _updateActivateSectorAssert(sectorNo sector.SectorNumber)
-
-    _getActiveSectorWeight()    (block.SectorWeight, error)
-    _getInactiveSectorWeight()  (block.SectorWeight, error)
-    _getPreCommitDepositReq()   actor.TokenAmount
-
-    _getSectorOnChainInfo(sectorNo sector.SectorNumber) (info SectorOnChainInfo, ok bool)
-    _getSectorPower(sectorNo sector.SectorNumber) (power block.StoragePower, ok bool)
-    _getSectorDealIDs(sectorNo sector.SectorNumber) (dealIDs deal.DealIDs, ok bool)
 }
 
 type StorageMinerActorCode struct {
-    GetOwnerKey(rt Runtime) filcrypto.VRFPublicKey
-    GetWorkerKey(rt Runtime) filcrypto.VRFPublicKey
-    OnCronTickEnd(rt Runtime)
-    NotifyOfSurprisePoStChallenge(rt Runtime)
-    ProcessSurprisePoSt(rt Runtime)
-
-    PreCommitSector(rt Runtime, info sector.SectorPreCommitInfo)  // TODO: check with Magik on sizes
-    ProveCommitSector(rt Runtime, info sector.SectorProveCommitInfo)
-    ProcessVerifiedElectionPoSt(rt Runtime)
-    DeclareFaults(rt Runtime, failingSet sector.CompactSectorSet)
-    RecoverFaults(rt Runtime, recoveringSet sector.CompactSectorSet)
-
-    _checkSurprisePoStSubmissionHappened(rt Runtime)
-    _rtVerifySurprisePoSt(rt Runtime, onChainInfo sector.OnChainPoStVerifyInfo) bool
-    _rtVerifySeal(rt Runtime, onChainInfo sector.OnChainSealVerifyInfo) bool
-    _rtAbortIfSectorDidNotExist(rt Runtime, sectorNo sector.SectorNumber)
-
-    _onMissedSurprisePoSt(rt Runtime)
-    _onSuccessfulPoSt(rt Runtime)
-
-    _slashCollateralForStorageFaults(
-        rt          Runtime
-        declared    sector.CompactSectorSet  // diff value
-        detected    sector.CompactSectorSet  // diff value
-        terminated  sector.CompactSectorSet  // diff value
-    )
-    _slashDealsForStorageFault(
-        rt             Runtime
-        sectorNumbers  [sector.SectorNumber]
-        faultType      sector.StorageFaultType
-    )
-
-    _submitPowerReport(rt Runtime, lastPoSt block.ChainEpoch)
-    _expirePreCommittedSectors(rt Runtime)
+    // Intentionally left blank (VM methods do not require Golang prototypes).
+    // Can be filled in once InvokeMethod is implemented for this actor.
 }
 
 type MinerInfo struct {
     // Account that owns this miner.
     // - Income and returned collateral are paid to this address.
     // - This address is also allowed to change the worker address for the miner.
     Owner                   address.Address  // Must be an ID-address.
-    OwnerKey                filcrypto.VRFPublicKey
 
     // Worker account for this miner.
     // This will be the key that is used to sign blocks created by this miner, and
     // sign messages sent on behalf of this miner to commit sectors, submit PoSts, and
     // other day to day miner activities.
-    Worker                  address.Address  // Must be an ID-address.
```

Comment:
Why did you remove this constraint? Note that the runtime immediate caller will always present an ID-address, so caller validation will fail unless this address matches.

Note that the key can be looked up from the address so the field below is unnecessary.

Reply:
You're right, this got lost in a merge conflict resolution -- I'll restore it in a new PR.

Reply:
#806